test/server/http_time_tracking_filter_integration_test.cc

```diff
@@ -0,0 +1,147 @@
+#include <chrono>
+
+#include "envoy/upstream/cluster_manager.h"
+#include "envoy/upstream/upstream.h"
+
+#include "external/envoy/test/common/upstream/utility.h"
+#include "external/envoy/test/integration/http_integration.h"
+#include "external/envoy/test/test_common/simulated_time_system.h"
+
+#include "api/server/response_options.pb.h"
+#include "api/server/response_options.pb.validate.h"
+
+#include "server/configuration.h"
+#include "server/http_time_tracking_filter.h"
+#include "server/well_known_headers.h"
+
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+
+using namespace std::chrono_literals;
+
+const std::string kLatencyResponseHeaderName = "x-prd";
+const std::string kDefaultProtoFragment = fmt::format(
+    "emit_previous_request_delta_in_response_header: \"{}\"", kLatencyResponseHeaderName);
+const std::string kProtoConfigTemplate = R"EOF(
+name: time-tracking
+typed_config:
+  "@type": type.googleapis.com/nighthawk.server.ResponseOptions
+  {}
+)EOF";
+
+class HttpTimeTrackingIntegrationTest
+    : public Envoy::HttpIntegrationTest,
+      public testing::TestWithParam<Envoy::Network::Address::IpVersion> {
+protected:
+  HttpTimeTrackingIntegrationTest()
+      : HttpIntegrationTest(Envoy::Http::CodecClient::Type::HTTP1, GetParam()),
+        request_headers_({{":method", "GET"}, {":path", "/"}, {":authority", "host"}}) {}
+
+  // We don't override SetUp(): tests in this file will call setup() instead to avoid having to
+  // create a fixture per filter configuration.
+  void setup(const std::string& config) {
+    config_helper_.addFilter(config);
+    HttpIntegrationTest::initialize();
+  }
+
+  // Fetches a response with request-level configuration set in the request header.
+  Envoy::IntegrationStreamDecoderPtr getResponse(absl::string_view request_level_config,
```

Comment:
There's a lot of overloading happening here that I think makes it a little hard to read.  Can we name these functions differently? Or alternatively, are all of them actually necessary?

In particular, the one with no arguments seems to counter the recommendation here: https://google.github.io/styleguide/cppguide.html#Function_Overloading

Reply:
I'd like to address this in a follow up; this is partially duplicated from other NH extensions, and I think this could be taken on as part of https://github.com/envoyproxy/nighthawk/issues/498.
Not having the code diverge makes it easier to refactor for now. Are you ok with that?

Reply:
Yep. That's completely reasonable.  Thanks!